{
  try {
    if (!processEnvelope(transferEnvelope)) {
      System.out.println("Processing envelope from network returned false");
    }
  }
 catch (  InterruptedException e) {
    LOG.error("Caught unexpected interrupted exception: " + StringUtils.stringifyException(e));
  }
  return false;
}
